src/widget/wlibrarytableview.cpp

```diff
@@ -164,15 +166,55 @@ bool WLibraryTableView::restoreTrackModelState(
     }
 
     QModelIndex currIndex = state->currentIndex;
-    if (currIndex.isValid()) {
-        selection->setCurrentIndex(currIndex, QItemSelectionModel::NoUpdate);
-    }
+    restoreCurrentIndex(currIndex);
 
     // reinsert the state into the cache
     m_modelStateCache.insert(key, state, 1);
     return true;
 }
 
+void WLibraryTableView::saveCurrentIndex() {
+    QItemSelectionModel* pSelectionModel = selectionModel();
+    if (!pSelectionModel) {
+        return;
+    }
+    QModelIndexList indices = pSelectionModel->selectedRows();
+    if (indices.isEmpty()) {
+        return;
+    }
+
+    m_prevRow = indices.first().row();
+    m_prevColumn = currentIndex().isValid() ? currentIndex().column() : columnAt(0);
+}
+
+void WLibraryTableView::restoreCurrentIndex(const std::optional<QModelIndex>& index) {
+    QItemSelectionModel* pSelectionModel = selectionModel();
+    if (!pSelectionModel) {
+        return;
+    }
+    int row = index ? index->row() : m_prevRow;
+    int col = index ? index->column() : m_prevColumn;
+    if (model()->rowCount() == 0 || row < 0 || col < 0) {
+        // nothing to select
+        return;
+    }
+    if (model()->rowCount() < row + 1) {
+        // select last row
+        row = model()->rowCount() - 1;
+    }
+    if (isColumnHidden(col)) {
+        // select first column
+        col = columnAt(0);
+    }
+    QModelIndex idx = model()->index(row, col);
+    if (idx.isValid()) {
+        pSelectionModel->setCurrentIndex(idx, QItemSelectionModel::NoUpdate);
```

Comment:
Does this mean we set only the current index, without actually selecting it? 
Do we need to actually select the row here in case index is valid e.g. after changing the playlist?
(Not tested yet)

Reply:
Jep, just setting the index here.
When being called from `restoreTrackModelState` that did restore the actual selection before.
When being called explicitly after track removal this is what you requested (and what makes sense, of course):
select tracks, remove > no selection left for clear feedback + starting point for navigation
Same when being called from `restoreTrackModelState` after changing the search phrase, when previously selected tracks may be gone.